src/cli/subcommands/mpool_cmd.rs

```diff
@@ -1,16 +1,408 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+use std::str::FromStr;
+
+use crate::json::cid::vec::CidJsonVec;
+use crate::json::signed_message::json::SignedMessageJson;
+use crate::message::{Message, SignedMessage};
+use crate::rpc_client::{chain_ops::*, mpool_pending, state_ops::*, wallet_ops::*};
+use crate::shim::{address::Address, econ::TokenAmount};
+
+use ahash::{HashMap, HashSet};
 use clap::Subcommand;
+use fvm_ipld_encoding::Cbor;
+use num::BigInt;
 
-use super::Config;
+use super::{handle_rpc_err, Config};
 
 #[derive(Debug, Subcommand)]
-pub enum MpoolCommands {}
+pub enum MpoolCommands {
+    /// Get pending messages
+    Pending {
+        /// Print pending messages for addresses in local wallet only
+        #[arg(long)]
+        local: bool,
+        /// Only print `CIDs` of messages in output
+        #[arg(long)]
+        cids: bool,
+        /// Return messages to a given address
+        #[arg(long)]
+        to: Option<String>,
```

Comment:
I wonder if could use a typed `Address` here We'd avoid handling it later in the code.

Reply:
I tried it, but did not managed because the conversion from `String` to `Address` can fail. If you have any clue I would be glad to hear about it.

Reply:
You could use `clap`s `parse` like here https://github.com/ChainSafe/forest/blob/293cb20d9f2b484a1c7ed1fefa23013679170e84/src/cli/subcommands/send_cmd.rs#L24-L25

Reply:
Unfortunately, differentiating between mainnet and testnet addresses requires parsing the addresses _after_ the command-line opts have been parsed.

Reply:
The `Address` design makes me sad, repeatedly.